_shared/project/README.md

```diff
@@ -0,0 +1,294 @@
+
+    ```terminal
+    make requirements
+    ```
+
+    ### To Upgrade or Downgrade a Dependency
+
+    We rely on [Dependabot](https://github.com/dependabot) to keep all our
+    dependencies up to date by sending automated pull requests to all our repos.
+    But if you need to upgrade or downgrade a package manually you can do that
+    locally.
+
+    To upgrade a package to the latest version in all `requirements/*.txt` files:
+
+    ```terminal
+    make requirements --always-make args='--upgrade-package <FOO>'
+    ```
+
+    To upgrade or downgrade a package to a specific version:
+
+    ```terminal
+    make requirements --always-make args='--upgrade-package <FOO>==<X.Y.Z>'
+    ```
+
+    To upgrade **all** packages to their latest versions:
+
+    ```terminal
+    make requirements --always-make args=--upgrade
+    ```
+{% endmacro %}
+
+{%- macro package_python_dependencies() %}
+    ## Changing the Project's Python Dependencies
+
+    To change the production dependencies in the `setup.cfg` file:
+
+    1. Change the dependencies in the [`.cookiecutter/includes/setuptools/install_requires`](.cookiecutter/includes/setuptools/install_requires) file.
+       If this file doesn't exist yet create it and add some dependencies to it.
+       For example:
+
+       ```
+       pyramid
+       sqlalchemy
+       celery
+       ```
+
+    2. Re-run the cookiecutter template:
+
+       ```terminal
+       make template
+       ```
+
+    3. Commit everything to git and send a pull request
+
+    To change the project's formatting, linting and test dependencies:
+
+    1. Change the dependencies in the [`.cookiecutter/includes/tox/deps`](.cookiecutter/includes/tox/deps) file.
+       If this file doesn't exist yet create it and add some dependencies to it.
+       Use tox's [factor-conditional settings](https://tox.wiki/en/latest/config.html#factors-and-factor-conditional-settings)
+       to limit which environment(s) each dependency is used in.
+       For example:
+
+       ```
+       lint: flake8,
+       format: autopep8,
+       lint,tests: pytest-faker,
+       ```
+
+    2. Re-run the cookiecutter template:
+
+       ```terminal
+       make template
+       ```
+
+    3. Commit everything to git and send a pull request
+{% endmacro -%}
+
+{{ badges()|dedent }}
+{{ intro()|dedent }}
+{% if cookiecutter.get("console_script") == "yes" -%}
+    {{ installing()|dedent }}
+{% endif %}
+{{ setting_up()|dedent }}
+{% if cookiecutter._directory == 'pypackage' -%}
+    {{ releasing()|dedent }}
+{% endif %}
+{% if cookiecutter._directory == 'pyramid-app' -%}
+    {{ app_python_version()|dedent }}
+{% else -%}
+    {{ package_python_version()|dedent }}
+{% endif %}
+{% if cookiecutter._directory == 'pyramid-app' -%}
+    {{ app_python_dependencies()|dedent }}
+{%- else -%}
+    {{ package_python_dependencies()|dedent }}
+{%- endif %}
+{% if include_exists("README/tail.md") %}
+
+{{ include("README/tail.md") -}}
+{% endif %}
```

Comment:
```suggestion
{{ badges() | dedent }}
{{ intro() | dedent }}
{% if cookiecutter.get("console_script") == "yes" -%}
    {{ installing() | dedent }}
{% endif %}
{{ setting_up() | dedent }}
{% if cookiecutter._directory == 'pypackage' -%}
    {{ releasing() | dedent }}
{% endif %}
{% if cookiecutter._directory == 'pyramid-app' -%}
    {{ app_python_version() | dedent }}
{% else -%}
    {{ package_python_version() | dedent }}
{% endif %}
{% if cookiecutter._directory == 'pyramid-app' -%}
    {{ app_python_dependencies() | dedent }}
{%- else -%}
    {{ package_python_dependencies() | dedent }}
{%- endif %}
{% if include_exists("README/tail.md") %}
    {{ include("README/tail.md") -}}
{% endif %}
```

Mostly putting some space around the pipe and indenting the final part.

Reply:
You can't indent that final `{{ include("README/tail.md") -}}` because there's more than one line in the body of the `{% if %}`: there's a blank line in the `{% if %}`-body above the `{{ include("README/tail.md") -}}`, the blank line is needed because we do want an empty line to appear in the output file before the contents of the `tail.md` file. Discussed elsewhere but generally speaking you can't indent lines in Jinja2 templates if you don't want those lines to be indented in the output. The custom `| dedent` filter here allows us to get away with indenting macro bodies but doesn't apply to blocks in general like `{% if %}` bodies.

I'm gonna leave the `|`'s without spaces around them as that's the style used in the Jinja2 docs